src/bashi/filter_software_dependency.py

```diff
@@ -116,4 +117,39 @@ def software_dependency_filter(
                 "older than 20.04",
             )
             return False
+
+    # Rule: d4
+    # Ubuntu 20.04 and newer is not available with CUDA older than 11
+
+    if UBUNTU in row and row[UBUNTU].version >= pkv.parse("20.04"):
+        if ALPAKA_ACC_GPU_CUDA_ENABLE in row and row[ALPAKA_ACC_GPU_CUDA_ENABLE].version != OFF_VER:
+            if row[ALPAKA_ACC_GPU_CUDA_ENABLE].version < pkv.parse("11.0"):
+                reason(
+                    output,
+                    f"CUDA {row[ALPAKA_ACC_GPU_CUDA_ENABLE].version} "
+                    "is not available in Ubuntu "
+                    f"{__ubuntu_version_to_string(row[UBUNTU].version)}",
+                )
+                return False
+        if DEVICE_COMPILER in row and row[DEVICE_COMPILER].name == NVCC:
+            if row[DEVICE_COMPILER].version < pkv.parse("11.0"):
+                reason(
+                    output,
+                    f"NVCC {row[DEVICE_COMPILER].version} "
+                    "is not available in Ubuntu "
+                    f"{__ubuntu_version_to_string(row[UBUNTU].version)}",
+                )
+                return False
+        for compiler_type in (HOST_COMPILER, DEVICE_COMPILER):
+            if compiler_type in row and row[compiler_type].name == CLANG_CUDA:
+                if row[compiler_type].version < get_oldest_supporting_clang_version_for_cuda(
+                    "11.0"
+                ):
+                    reason(
+                        output,
+                        f"{__pretty_name_compiler(compiler_type)} clang cuda {row[compiler_type].version} "
```

Comment:
```suggestion
                        f"{__pretty_name_compiler(compiler_type)} clang-cuda {row[compiler_type].version} "
```
Just a little bit sugar ;-)